doc-bridgepoint/notes/8769_opaque_components.int.md

```diff
@@ -0,0 +1,131 @@
+---
+
+This work is licensed under the Creative Commons CC0 License
+
+---
+
+# Make components opaque
+### xtUML Project Implementation Note
+
+1. Abstract
+-----------
+Components hide functionality and implementation details from external elements
+except by accessing ports. However, elements outside a component had been visible
+from the inside. This property could cause unwanted consequences when package
+references are introduced. This work is to allow components to restrict
+visibility of external elements as well.
+
+2. Document References
+----------------------
+<a id="2.1"></a>2.1 [#8769 Within a component restrict visibility to externally defined model elements](https://support.onefact.net/issues/8769)  
+Parent issue of this work.
+
+<a id="2.2"></a>2.1 [#8798 The "cut" currently implemented (Model Element Move) allows a limited element selection](https://support.onefact.net/issues/8798)  
+
```

Comment:
I think you need to add references to the document and truth table John wrote that defines the visibility plans.  They are fundamental to this work.